SwiftLeeds/Views/My Conference/MyConferenceViewModel.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  MyConferenceViewModel.swift
+//  SwiftLeeds
+//
+//  Created by Matthew Gallagher on 01/08/2022.
+//
+
+import Foundation
+import Combine
+import NetworkKit
+
+class MyConferenceViewModel: ObservableObject {
+    @Published var event: Schedule.Event?
+    @Published var slots: [Schedule.Slot] = []
+
+    @MainActor
+    func loadSchedule() async throws {
```

Comment:
Regardless of the underlying network implementation, it's not ideal to have this whole function on Main.

Let's split this up, so the load function isn't MainActor, it instead hits the network and then calls a private function ( say publishResults(slots) ) which is MainActor. That way you're only on main for the part you absolutely need to be.

An alternative would be that you don't use Published, and after the network call simply do MainActor.run { objectWillChange.send() }. Given we're in full control of the reloads here that could be a good choice 🤔

Reply:
This is following how I've seen it working, so will look more into it but leave it like this until I have more understanding on the subject.

Reply:
It's not a very collaborative way to work if we refuse to change things. 

PRs are a place for everyone to learn and improve, not to throw code in and just "leave it until I have more understanding". If we don't understand something, we simply shouldn't use it, or should be open to help from those who have. I understand it's tempting to use the latest and greatest, but if we don't know what's going on that's dangerous.

I did offer an explanation as to why it's not a good idea, and would be happy to re-explain if needed, wether here or via a pair.

Ultimately this is running all the code within this function on main, which is not a good idea. Regardless of your understanding, it's simply bad practice.

There's a bunch of simpler examples here that tell you when things are running on main. You wouldn't add DispatchQueue.main to your networking functions, so you also wouldn't add MainActor. Same principle, different syntax.

https://www.hackingwithswift.com/quick-start/concurrency/how-to-use-mainactor-to-run-code-on-the-main-queue

Reply:
Alex, you refuse to change things on your PR's all the time so maybe learn from your own statement.  
You are reviewing code, not ordering the person doing the code to change things.  The implementation is correct from my understanding, so happy with it, but at the same time will take some time to understand if there is a better way to improve it in the future, but not going to change it just because you said so.  

Would be a lot more helpful if you used PR's to actual review the code, and not add a rant.

Reply:
This is very unfortunate.

I don't believe this is a productive collaboration, and has simply devolved into trying to win. 

Simply check what thread your code is running on in the debugger to validate what I’m talking about here - it’s not opinion, it’s nothing to get angry about, it’s just learning something new. If you’re not sure on debugging that way, the link conditioner set to edge will illustrate the behaviour too.

I’m going to remove myself from this pull request.